src/services/OfferService.js

```diff
@@ -84,6 +84,9 @@ export default class OfferService {
       .lean()
       .exec()
 
+    // Can't exclude the city in the projection because of populate
+    offers.forEach(offer => delete offer.city)
```

Comment:
Can you do another select after the populate? (Just curious^^)

Reply:
Tried it, but didn't work (for me). I think the thing is that mongoose is just submitting the query to the mongodb and mongodb deals with the exact order of execution.